vm/src/stdlib/os.rs

```diff
@@ -555,8 +555,30 @@ mod _os {
     }
 
     #[pyfunction]
-    fn listdir(path: PyPathLike, vm: &VirtualMachine) -> PyResult {
-        let dir_iter = fs::read_dir(&path.path).map_err(|err| err.into_pyexception(vm))?;
+    fn listdir(path: Either<PyPathLike, i64>, vm: &VirtualMachine) -> PyResult {
+        let path = match path {
+            Either::A(path) => path,
+            Either::B(fno) => {
+                cfg_if::cfg_if! {
+                    if #[cfg(any(target_os = "linux", target_os = "macos", windows))] {
+                        let file = rust_file(fno);
+                        match file.path() {
+                            Ok(path) => PyPathLike::new_str(path.to_string_lossy().to_string()),
+                            Err(_) => {
+                                return Err(vm.new_os_error(format!(
+                                    "Cannot determine path of fno: {:?}",
+                                    fno
+                                )));
+                            }
+                        }
+                    } else {
+                        return Err(vm.new_os_error("dir_fd not supported on wasi yet".to_owned()));
+                    }
+                }
```

Comment:
I think you could just call `make_path` here - then for the other, do `path.path.into()` to convert it to a `Cow<Path>`